v3/src/components/graph/adornments/lsrl/lsrl-adornment-component.tsx

```diff
@@ -0,0 +1,359 @@
+      // with the line's mouseover behavior.
+      lineObj.confidenceBandCurve = selection.append("path")
+        .attr("class", `lsrl-confidence-band lsrl-confidence-band-${classFromKey}`)
+        .attr("data-testid", `lsrl-confidence-band${classFromKey ? `-${classFromKey}` : ""}`)
+      lineObj.confidenceBandCover = selection.append("path")
+        .attr("class", `lsrl-confidence-band-cover lsrl-confidence-band-cover-${classFromKey}`)
+        .attr("data-testid", `lsrl-confidence-band-cover${classFromKey ? `-${classFromKey}` : ""}`)
+      lineObj.confidenceBandShading = selection.append("path")
+        .attr("class", `lsrl-confidence-band-shading lsrl-confidence-band-shading-${classFromKey}`)
+        .attr("data-testid", `lsrl-confidence-band-shading${classFromKey ? `-${classFromKey}` : ""}`)
+
+      // Set up the line and its cover element
+      lineObj.line = selection.append("line")
+        .attr("class", `lsrl lsrl-${classFromKey}`)
+        .attr("data-testid", `lsrl${classFromKey ? `-${classFromKey}` : ""}`)
+      lineObj.cover = selection.append("line")
+        .attr("class", `lsrl-cover lsrl-cover-${classFromKey}`)
+        .attr("data-testid", `lsrl-cover${classFromKey ? `-${classFromKey}` : ""}`)
+      lineObj.cover.on("mouseover", () => { handleHighlightLineAndEquation(true, lineIndex) })
+        .on("mouseout", () => { handleHighlightLineAndEquation(false, lineIndex) })
+
+      // Set the line's coordinates
+      fixEndPoints(lineObj?.line)
+      fixEndPoints(lineObj?.cover)
+
+      // If there is a categorical legend, use the associated category's color for the line and its confidence bands
+      catColor && lineObj?.line?.style("stroke", catColor)
+      catColor && lineObj?.confidenceBandCurve?.style("stroke", catColor)
+      catColor && lineObj?.confidenceBandShading?.style("fill", catColor)
+    
+      // Add the equation box for the line to the equation container
+      const equationP = equationDiv
+        .append("p")
+        .attr("class", "lsrl-equation")
+        .attr("data-testid", `lsrl-equation-${model.classNameFromKey(cellKey)}`)
+        .on("mouseover", () => { handleHighlightLineAndEquation(true, lineIndex) })
+        .on("mouseout", () => { handleHighlightLineAndEquation(false, lineIndex) })
+
+      // If the equation is not pinned to the line, set its initial coordinates to the values specified in the model.
+      const equationCoords = lines[lineIndex]?.equationCoords
+      if (equationCoords?.isValid()) {
+        const left = equationCoords.x * 100
+        const top = equationCoords.y * 100
+        equationP.style("left", `${left}%`)
+          .style("top", `${top}%`)
+      }
+
+      lineObj.equation = equationDiv
+      lineObjectsRef.current = [...lineObjectsRef.current, lineObj]
+
+      const equation = equationDiv.selectAll("p").filter((d, i) => i === lineIndex)
+      equation.call(
+        drag<any, any>().on("drag", (e) => handleMoveEquation(e, false, lineIndex))
```

Comment:
I did try to find more specific types to use here, but had trouble and decided not to spend anymore time on it for now.